munch/__init__.py

```diff
@@ -259,6 +263,42 @@ def __repr__(self):
             type(self).__name__, self.__undefined__, dict.__repr__(self))
 
 
+class DefaultFactoryMunch(defaultdict, Munch):
+    """ A Munch that calls a user-specified function to generate values for
+        missing keys like collections.defaultdict.
+
+        >>> b = DefaultFactoryMunch(list, {'hello': 'world!'})
+        >>> b.hello
+        'world!'
+        >>> b.foo
+        []
+        >>> b.bar.append('hello')
+        >>> b.bar
+        ['hello']
+    """
+    def __init__(self, *args, **kwargs):
+        if not args:
+            def default():
+                return None
+
+            args = [default]
+
+        super(DefaultFactoryMunch, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def fromDict(cls, d, default=lambda: None):
```

Comment:
This makes no sense - what is the use case in which `default` won't be passed???

Reply:
I was copying the semantics of DefaultMunch by providing a default value. I can remove it here (and in __init__)